### ISSUE TITLE
Running terraform validate with a terraform_remote_state including a role_arn config causes validate to fail with cannot assume role

### ISSUE THREAD

**Issue by graemedavidson** [hashicorp/terraform#22354]
Assumptions/expectations would be that this would not be required as part of the validate function.

### Terraform Version
```
v0.12.6
```

### Terraform Configuration Files
```hcl
data "terraform_remote_state" "remote_state" {
    backend = "s3"
                                                                                                                                                                                                                                                                                       
    config = {
        bucket   = "<bucket>"
        key      = "<key>"
        region   = "region"
        role_arn = "<role_arn>"
    }
}
```

### Debug Output
2019/08/06 15:43:40 [INFO] Attempting to AssumeRole arn:aws:iam::#######:role/###/### (SessionName: "", ExternalId: "", Policy: "")                                                                                                                                           
2019/08/06 15:43:40 [INFO] AWS Auth provider used: "SharedCredentialsProvider"
2019/08/06 15:43:40 [WARN] <root>: eval: *terraform.EvalValidateResource, non-fatal err: The role "arn:aws:iam::####:role/###/####" cannot be assumed.                                                                                                                      

### Expected Behavior
Success! The configuration is valid.

### Actual Behavior
Error: The role "arn:aws:iam::####:role/###/####" cannot be assumed.

  There are a number of possible causes of this - the most common are:
    * The credentials used in order to assume the role are invalid
    * The credentials do not have appropriate permission to assume the role
    * The role ARN is not valid

### Steps to Reproduce
1. `terraform init -backend=false -get=true`
2. `terraform validate`

Potentially linked to this ticket #21761 but seems to be a different issue hence posting as separate issue

**Comment by ktham** [comment]
Any updates? My workspaces that contain a `terraform_remote_state` data resource cannot pass `terraform validate` even after `terraform init -backend=false`

I see
```Error: Required token could not be found

Make sure you configured a credentials block for app.terraform.io in your CLI
Config File.
```

**Comment by AYapejian** [comment]
Found a workaround for this and may shed some light for someone to fix:

In both cases below the following was done:

1. Create `main.tf` with contents of below block
2. `rm -rf .terraform && terraform init --backend=false`
3. `terraform validate`

### Failing terraform validate in v0.12.23 ( which worked in v0.11.14 )

```terraform
terraform {
  required_version = "0.12.23"

  backend "s3" {
    region         = "us-east-1"
    bucket         = "sometfstatebucket"
    key            = "mytfstate/terraform.tfstate"
    role_arn       = "arn:aws:iam::012345678901:role/TFStateAdmin"
    encrypt        = true
    dynamodb_table = "TerraformDeployment"
  }
}

provider "aws" {
  region = "us-east-1"

  assume_role {
    role_arn = "arn:aws:iam::012345678901:role/TFUser"
  }

  version = "~> 2.52"
}

data "terraform_remote_state" "remotestate_1" {
  backend = "s3"
  config  = {
    region         = "us-east-1"
    bucket         = "sometfstatebucket"
    key            = "tfstatetest/terraform.tfstate"
    role_arn       = "arn:aws:iam::012345678901:role/TFStateUser"
    encrypt        = true
    dynamodb_table = "TerraformDeployment"
  }
}
```

### Workaround which allows terraform validate without assuming roles of remote state in v0.12.23

The only changes here are to externalize the `terraform_remote_state.remotestate_1.role_arn` to a `variable` and use that instead of the hardcoded arn.  This allows validate to work without trying to assume the `role_arn` role after init with `--backend=false`

```terraform
terraform {
  required_version = "0.12.23"

  backend "s3" {
    region         = "us-east-1"
    bucket         = "sometfstatebucket"
    key            = "mytfstate/terraform.tfstate"
    role_arn       = "arn:aws:iam::012345678901:role/TFStateAdmin"
    encrypt        = true
    dynamodb_table = "TerraformDeployment"
  }
}

provider "aws" {
  region = "us-east-1"

  assume_role {
    role_arn = "arn:aws:iam::012345678901:role/TFUser"
  }

  version = "~> 2.52"
}

variable "remotestate_assume_role" {
  default = "arn:aws:iam::012345678901:role/TFStateUser"
}

data "terraform_remote_state" "remotestate_1" {
  backend = "s3"
  config  = {
    region         = "us-east-1"
    bucket         = "sometfstatebucket"
    key            = "tfstatetest/terraform.tfstate"
    role_arn       = var.remotestate_assume_role
    encrypt        = true
    dynamodb_table = "TerraformDeployment"
  }
}
```

**Comment by graemedavidson** [comment]
This is still an issue in tf 0.12 but is resolved in 0.13 and above.
